src/client/gui/lib/settings/autostart_notifiers.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final autostartProvider =
+    AsyncNotifierProvider.autoDispose<AutostartNotifier, bool>(() {
+  if (Platform.isLinux) return LinuxAutostartNotifier();
+  if (Platform.isWindows) return WindowsAutostartNotifier();
+  if (Platform.isMacOS) return MacOSAutostartNotifier();
```

Comment:
Can we encapsulate any platform dependent code like this, as we do in C++?

Reply:
Not quite as in C++. IIUC, in C++ we decide at build time which platform implementation to use. But that is not possible in Flutter. We have to do the check at runtime and return our desired implementation based on that.

Reply:
OK, that is fine, but what I meant is to have any platform specific code - such as these `if`s and the concrete `AutostartNotifier`s - in a a separate file/folder, together with any other platform specific code. We then know that everything else is common to all platforms and that, if we we're looking for platform code, that is the only folder we need to be concerned with. 

We always strived for this organization in the Multipass repo and it would be a shame to lose it now. Fortunately you already encapsulated platform behaviors in dedicated types, so it should be pretty easy to move.